### ISSUE TITLE
System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to

### ISSUE THREAD

**Issue by Romfos** [microsoft/vstest#4671]
Hi,

MSTest v3 Stop working for linux

Nuget:
```
<PackageReference Include="MSTest" Version="3.1.1" />
```
Build link link:
https://github.com/Romfos/AutoTests.Framework/actions/runs/5866367920/job/15905047692

```
Testhost process for source(s) '/home/runner/work/AutoTests.Framework/AutoTests.Framework/src/AutoTests.Framework.Tests/bin/Debug/net462/AutoTests.Framework.Tests.dll' exited with error: Unhandled Exception:
System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e0[89](https://github.com/Romfos/AutoTests.Framework/actions/runs/5866367920/job/15905047692#step:5:90)') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
   --- End of inner exception stack trace ---
  at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main (System.String[] args) [0x00033] in <af8fc7b1c76049579919cce3f8e01556>:0 
Error:  FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
   --- End of inner exception stack trace ---
  at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main (System.String[] args) [0x00033] in <af8fc7b1c76049579[91](https://github.com/Romfos/AutoTests.Framework/actions/runs/5866367920/job/15905047692#step:5:92)9cce3f8e01556>:0 
. Please check the diagnostic logs for more information.
```

**Comment by Evangelink** [comment]
I have transfered the issue to vstest repo because I have confirmed issue is linked to it. Uninstalling .NET 7 fixes the issue.

**Comment by nohwnd** [comment]
This is probably the same issue as #4549 and #4627, we inserted fix for that into servicing on 7.0.401, and it should be in the next service release.  (this has links to the insertion pr https://github.com/microsoft/vstest/issues/4549#issuecomment-1662494569)
